expression/builtin.go

```diff
@@ -37,6 +38,8 @@ type baseBuiltinFunc struct {
 	ctx    sessionctx.Context
 	tp     *types.FieldType
 	pbCode tipb.ScalarFuncSig
+
+	vectorizedAllFlag atomic.Uint32 // 0 unknown, 1 vectorized, 2 not vectorized
```

Comment:
We may use the singleton pattern here to set the value for vectorizedAllFlag?

Reply:
You mean that we can use the `sync.Once`?

Reply:
I think `sync.Once` is ok.

Reply:
fixed.